R/extract.R

```diff
@@ -7,6 +7,33 @@
 setGeneric("extract", function(model, ...) standardGeneric("extract"), 
     package = "texreg")
 
+# default extract method prompts users to install the broom package
+extract.default <- function(model, ...) {
+  if (!'broom' %in% row.names(installed.packages())) {
+    stop("texreg does not directly support models of class ",
+         class(model), 
+         ", but it can sometimes use the ``broom`` library to extract model information. Call texreg again after installing the broom library to see if this is possible.")
+  }
+  estimates <- try(broom::tidy(model), silent = TRUE)
+  gof <- try(broom::glance(model), silent = TRUE)
+  if ((class(estimates) == 'try-error') || 
+      (class(gof) == 'try-error') ||
+      !all(c('term', 'estimate', 'std.error', 'p.value') %in% colnames(estimates))) { # tidy sometimes produces "non-standard" results
+    stop('Neither texreg nor broom supports models of class ', class(model), '.')
+  }
+  tr <- createTexreg(coef.names = estimates$term, 
+                     coef = estimates$estimate, 
+                     se = estimates$std.error, 
+                     pvalues = estimates$p.value, 
+                     gof.names = colnames(gof), 
+                     gof = as.numeric(gof[1, ]), 
+                     gof.decimal = rep(TRUE, ncol(gof)) 
```

Comment:
The `gof.decimal` slot does not have to be identical across models for the same model term. See the following example:
```
library("texreg")

# example from ?lm
ctl <- c(4.17,5.58,5.18,6.11,4.50,4.61,5.17,4.53,5.33,5.14)
trt <- c(4.81,4.17,4.41,3.59,5.87,3.83,6.03,4.89,4.32,4.69)
group <- gl(2, 10, 20, labels = c("Ctl","Trt"))
weight <- c(ctl, trt)
lm.D9 <- lm(weight ~ group)
lm.D90 <- lm(weight ~ group - 1) # omitting intercept

model1 <- extract(lm.D9)
model2 <- extract(lm.D90)
model2@gof.decimal[3] <- TRUE

screenreg(list(model1, model2))
```
which produces this table:
```
=================================
             Model 1    Model 2  
---------------------------------
(Intercept)   5.03 ***           
             (0.22)              
groupTrt     -0.37       4.66 ***
             (0.31)     (0.22)   
groupCtl                 5.03 ***
                        (0.22)   
---------------------------------
R^2           0.07       0.98    
Adj. R^2      0.02       0.98    
Num. obs.    20         20.00    
RMSE          0.70       0.70    
=================================
*** p < 0.001, ** p < 0.01, * p < 0.05
```
Note the different decimal settings for `Num. obs`.

So if you can check in each case if the entry is integer, that should account for exceptions in other models.